src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol

```diff
@@ -55,104 +94,386 @@ contract BancorVirtualSupplyBondingCurveFundingManager is
+        // Set reserve ratio for selling
+        _setReserveRatioForSelling(_reserveRatioForSelling);
+        // Set buy fee percentage
+        _setBuyFee(_buyFee);
+        // Set sell fee percentage
+        _setSellFee(_sellFee);
+        // Set buying functionality to open if true. By default buying is false
+        if (_buyIsOpen == true) _openBuy();
+        // Set selling functionality to open if true. By default selling is false
+        if (_sellIsOpen == true) _openSell();
     }
 
     //--------------------------------------------------------------------------
     // Public Functions
 
+    /// @notice Buy tokens on behalf of a specified receiver address. This function is subject
+    /// to a transactional limit, determined by the deposit token's decimal precision and the underlying
+    /// bonding curve algorithm.
+    /// @dev Redirects to the internal function `_buyOrder` by passing the receiver address and deposit amount.
+    /// Important: The Bancor Formula has an upper computational limit of (10^38). For tokens with
+    /// 18 decimal places, this effectively leaves a maximum allowable deposit amount of (10^20).
+    /// While this is substantially large, it is crucial to be aware of this constraint.
+    /// Transactions exceeding this limit will be reverted.
+    /// @param _receiver The address that will receive the bought tokens.
+    /// @param _depositAmount The amount of collateral token depoisited.
+    function buyOrderFor(address _receiver, uint _depositAmount)
+        external
+        payable
+        override(BondingCurveFundingManagerBase)
+        validReceiver(_receiver)
+        buyingIsEnabled
+    {
+        _virtualSupplyBuyOrder(_receiver, _depositAmount);
+    }
+
+    /// @notice Buy tokens for the sender's address. This function is subject
+    /// to a transactional limit, determined by the deposit token's decimal precision and the underlying
+    /// bonding curve algorithm.
+    /// @dev Redirects to the internal function `_buyOrder` by passing the sender's address and deposit amount.
+    /// Important: The Bancor Formula has an upper computational limit of (10^38). For tokens with
+    /// 18 decimal places, this effectively leaves a maximum allowable deposit amount of (10^20).
+    /// While this is substantially large, it is crucial to be aware of this constraint.
+    /// Transactions exceeding this limit will be reverted.
+    /// @param _depositAmount The amount of collateral token depoisited.
     function buyOrder(uint _depositAmount)
         external
         payable
         override(BondingCurveFundingManagerBase)
+        buyingIsEnabled
     {
-        // WiP
-        // Deduct fee from incoming value. Fee is paid in collateral token
-        uint amountIssued = _issueTokens(_depositAmount, collateral);
-        _addTokenAmount(amountIssued);
-        _addCollateralAmount(_depositAmount);
+        _virtualSupplyBuyOrder(_msgSender(), _depositAmount);
```

Comment:
Not sure about this one... It does create a buy order with the virtual supply, and not the actual one. I created the function to avoid code duplication. Of course we can also make _buyOrder virtual and override it, but to me this seems more explicit?

Reply:
maybe renaming to virtualBuyOrder is better ?

Reply:
I was a bit confused, when i read it, because you combine the two of them together basically, but the name suggests that you use just the virtual

Reply:
I think the idea is that the base contract has the `buyOrder` function of course. But because our implementation makes use of a virtual supply, we prepend the virtual supply part to the buyOrder.

I think overriding _buyOrder would defeat the purpose of our abstract formula wrappers

Reply:
But arent you overriding it currently anyway?
I think for me it was more of a design thing anyway.
Because your using the original _buyOrder function anyway this doesnt make so much of a difference,
but I thought because you extend it anyway you could also write it the way I described it before